tests/test_photoCalib.py

```diff
@@ -363,6 +365,49 @@ def testCalibEquality(self):
 
         self.assertFalse(photoCalib1 != photoCalib2)  # using assertFalse to directly test != operator
 
+    def testCalibrateImage(self):
+        dim = (5, 6)
+        npDim = (dim[1], dim[0])  # numpy and afw have a different x/y order
+        sigma = 10.0
+        image = np.random.normal(loc=1000.0, scale=sigma, size=npDim).astype(np.float32)
+        mask = np.zeros(npDim, dtype=np.int32)
+        variance = (np.random.normal(loc=0.0, scale=sigma, size=npDim).astype(np.float32))**2
+        maskedImage = lsst.afw.image.basicUtils.makeMaskedImageFromArrays(image, mask, variance)
+        maskedImage.mask[0, 0] = True  # set one mask bit to check propagation of mask bits.
+
+        def makeExpected(outImage, calibration):
+            outErr = computeMaggiesErr(np.sqrt(variance),
+                                       image,
+                                       self.calibrationErr,
+                                       calibration,
+                                       outImage).astype(np.float32)  # variance plane must be 32bit
+            return lsst.afw.image.makeMaskedImageFromArrays(outImage,
+                                                            maskedImage.mask.getArray(),
+                                                            outErr**2)
```

Comment:
Why not move this to outside `testCalibrateImage` (with a more descriptive name, of course)? I find nested function definitions hard to read.